akka-stream/src/main/scala/akka/stream/impl/UnfoldResourceSource.scala

```diff
@@ -47,6 +47,7 @@ import scala.util.control.NonFatal
         case NonFatal(ex) ⇒ decider(ex) match {
           case Supervision.Stop ⇒
             close(blockingStream)
+            open = false
```

Comment:
I think there can still be two invocations if `close` throws, because then `open` is still true in `postStop`. Would be better to set `open = false` before calling `close. Same problem in `def closeStage` and perhaps also in `def restartState`

Reply:
Good point, I'll move it up

Reply:
Heh, if I move this place, the other exception-on-close test starts failing, and I must be stupid or something, because I cannot understand why.

The other place was fine to move around and fixed a reproducing test case I wrote.